### ISSUE TITLE
Generalization Ability of the Operator

### ISSUE THREAD

**Issue by MIA123MIA321** [zongyi-li/fourier_neural_operator#86]
Hi,

Thanks for your great idea of deleting the high-frequency terms. Recently, I am trying to use FNO to learn a PDE operator, which is based on the inverse medium scattering problem. I am faced with some problems. I'd be grateful to receive your advice.

The PDE is described as followed:
$\Delta u + k^2(1+q)u = -k^2qf \quad \text{in} \quad \Omega , \qquad\frac{\partial u}{\partial \nu} = iku \quad \text{on} \quad {\partial \Omega}$
where $\Omega$ is taken as $[0,1]^2$, $k$ is a fixed frequency, $q>-1$ is the perturbation term, and the right term $-k^2qf$ is **fixed as 1** at present.
As a result, we want to learn the nonlinear operator from $q$ to $u$.

At first, I generate $q$ using the method in your code, 'lognorm' or the 'thresh'. The generating method of $q$ on the training set and on the testing set are the same. To my delight, the network trained on the train set can get accurate solutions on the test set. But when I change the generating method of $q$, it does not work on solving the PDE.

Because of the ill-posedness of this PDE, $u$ can be affected by the small change of q, either the shape or the size. 
For example, after the set of $q$ with the maximum of 0.1 is taken as the training set, it does not perform well on the set of $10q$ with the maximum of 1.

Can you offer me some advice on training the network to solve the PDE as a substitute for traditional methods, no matter what the perturbation $q$ is like?

**Comment by zongyi-li** [comment]
Dear Ziyang,

Thanks for your interest in our work. This is indeed a very good question. Indeed, the extrapolation or domain shift problem is still one of the common challenges for data-driven methods with no clear answer. There are a few perspectives:

- For practical use, we can usually decide on the training dataset while the testing case may be unknown. It's better to generate a widely representative distribution, and the training dataset is harder than the testing case. For example, you can generate the dataset with q with a larger value (say, maximum of 10) which gives you a stronger operator. Posting a limitation on the training data inevitably limits the power of the data-driven model.
- Hybrid model: instead of learning the full solution operator, we can learn an enhancing operator based on a coarse solver or simpler poison solver.
- Physics-informed neural operator: as the equation is given, it's possible to apply an equation loss in the test time.
